### PR TITLE
API with more options

### PR DESCRIPTION
**Pull request by differentreality**
includes #712

### CODE REVIEW

File: app/serializers/conference_serializer.rb [hunk 1]
```diff
@@ -20,16 +68,18 @@ def revision
...
-  # FIXME: just giving suseconferenceclient something to play with
-  def description
-    'openSUSE Conference 2013 - Power to the Geeko'
-  end
-
-  # FIXME: same than the former
-  def url
-    'https://conference.opensuse.org/'
-  end
+#   # FIXME: just giving suseconferenceclient something to play with
```
Comment (hennevogel) [inline app/serializers/conference_serializer.rb]: Why commented this stuff?

File: app/controllers/api/v1/rooms_controller.rb [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
       respond_to :json
 
       def index
-        if params[:conference_id].blank?
-          rooms = Room.all
-        else
-          conference = Conference.find_by_guid(params[:conference_id])
-          rooms = conference.rooms
-        end
+        @conference ? (rooms = @conference.rooms) : (rooms = Room.all)
```
Comment (hennevogel) [inline app/controllers/api/v1/rooms_controller.rb]: I wonder if this makes sense. Listing all the rooms in the DB...

File: app/controllers/api/v1/tracks_controller.rb [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
       respond_to :json
 
       def index
-        if params[:conference_id].blank?
-          tracks = Track.all
-        else
-          tracks = Track.joins(:conference)
-          tracks = tracks.where(conferences: { guid: params[:conference_id] })
-        end
+        @conference ? (tracks = @conference.tracks) : (tracks = Track.all)
```
Comment (hennevogel) [inline app/controllers/api/v1/tracks_controller.rb]: Same here. Who could be interested in all tracks in the DB?

File: app/serializers/conference_serializer.rb [hunk 4]
```diff
@@ -20,16 +68,18 @@ def revision
   # FIXME: adjusting the format the DIRTY way, for oSC13.
   # If you think this is ugly, don't look at the methods below
```
Comment (hennevogel) [inline app/serializers/conference_serializer.rb]: I doubt this comment has any meaning anymore...

File: app/serializers/conference_serializer.rb (hunk 1 above)
Comment (differentreality) [inline app/serializers/conference_serializer.rb]: Ctrl+D instead of Ctrl+S I guess :)

File: app/controllers/api/v1/rooms_controller.rb (hunk 2 above)
Comment (differentreality) [inline app/controllers/api/v1/rooms_controller.rb]: It was possible to list all rooms before, so if there is any chance someone used that data to do sth, if we remove it now, things will break.

I think there is small possibility that someone might need that info. The only scenario I can think of is multiple conferences being co-hosted, 1 OSEM instance, but they want to extract overall data for all conferences to do sth with that data.

I'd more focus on what's the harm keeping it?

**Comment by differentreality** [comment]
Ready for review!

**Comment by ChrisBr** [comment]
LGTM if @hennevogel is okay with it!

File: app/controllers/api/v1/rooms_controller.rb (hunk 2 above)
Comment (hennevogel) [inline app/controllers/api/v1/rooms_controller.rb]: Cruft that nobody needs

**Comment by hennevogel** [comment]
Anyway. Let's get this in :-)

**Pull request merged**